packages/paste-website/src/pages/components/editable-code-block/index.mdx

```diff
@@ -0,0 +1,370 @@
+    height="45vh"
+    defaultLanguage="typescript"
+    defaultValue={TypeScriptExample}
+  />
+</EditableCodeBlockWrapper>`}
+/>
+
+### Simple Editable Code Block
+
+Remove line numbers, indentation guides, and code folding from Editable Code Block only when:
+
+- Your page layout makes it visually clear to the user that the code block is editable, _and_
+- The expected code input is simple, like a short snippet of JSON.
+
+<StoryPreview
+  title="Simple Editable Code Block"
+  storyID="components-editable-code-block--no-special-features"
+  code={`<EditableCodeBlockWrapper>
+  <EditableCodeBlockHeader>Simple Editable Code Block</EditableCodeBlockHeader>
+  <EditableCodeBlock
+    height="45vh"
+    defaultLanguage="typescript"
+    lineNumbers="off"
+    folding={false}
+    indentationGuide={false}
+    defaultValue={TypeScriptExample}
+  />
+</EditableCodeBlockWrapper>`}
+/>
+
+### Editable Code Block with minimap
+
+For large files, use a minimap to help users navigate the code block.
+
+<StoryPreview
+  title="Editable Code Block with minimap"
+  storyID="components-editable-code-block--minimap"
+  code={`<EditableCodeBlockWrapper>
+  <EditableCodeBlockHeader>Minimap example</EditableCodeBlockHeader>
+  <EditableCodeBlock height="45vh" defaultLanguage="typescript" showMinimap defaultValue={TypeScriptExample} />
+</EditableCodeBlockWrapper>`}
+/>
+
+### Multi-file Editable Code Block
+
+You can alternate the content in a single instance of the Editable Code Block.
+This is useful for editing multiple files in a single editor.
+
+<StoryPreview
+  title="Multi-file Editable Code Block"
+  storyID="components-editable-code-block--multi-file-editor"
+  code={`<Stack orientation="vertical" spacing="space40">
+  <RadioButtonGroup
+    attached
+    name="file-selector"
+    legend="Select file"
+    helpText="Show code for the selected file"
+    onChange={(_fileKey) => setFileKey(_fileKey)}
+  >
+    {Object.keys(Files).map((key) => (
+      <RadioButton key={key} value={key} defaultChecked={fileKey === key}>
+        {Files[key].name}
+      </RadioButton>
+    ))}
+  </RadioButtonGroup>
+  <EditableCodeBlockWrapper>
+    <EditableCodeBlockHeader>{Files[fileKey].name}</EditableCodeBlockHeader>
+    <EditableCodeBlock
+      height="400px"
+      path={Files[fileKey].name}
+      defaultLanguage={Files[fileKey].language}
+      defaultValue={Files[fileKey].value}
+    />
+  </EditableCodeBlockWrapper>
+</Stack>`}
+/>
+
+## States
+
+### Read only
+
+If the user doesn't have permission to edit the code, use the `readOnly` prop to make the Editable Code Block read-only.
+
+<StoryPreview
+  title="Read-Only Editable Code Block"
+  storyID="components-editable-code-block--read-only"
+  code={`<EditableCodeBlockWrapper>
+  <EditableCodeBlockHeader>ReadOnly Example</EditableCodeBlockHeader>
+  <EditableCodeBlock height="45vh" readOnly defaultLanguage="typescript" defaultValue={TypeScriptExample} />
+</EditableCodeBlockWrapper>`}
+/>
```

Comment:
suggestion: Move this underneath the error examples since it's a less-common use case